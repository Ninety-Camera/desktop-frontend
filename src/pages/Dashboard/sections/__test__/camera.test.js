import React from "react";
import { render, screen, cleanup } from "@testing-library/react";
import ReactDom from "react-dom";
import Camera from "../camera";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";
import { useState } from "react";
import { act } from "react-dom/test-utils";

afterEach(cleanup);

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDom.render(<Camera />, div);
});

it("matches snapshot", () => {
  const tree = renderer.create(<Camera />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("Toggle Button changes values properly", ()=>{
    const onChange = jest.fn();
    act(()=>{
        
    });
});