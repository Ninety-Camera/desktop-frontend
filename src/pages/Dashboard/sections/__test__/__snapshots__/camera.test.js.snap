// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  style={
    Object {
      "overflow": "hidden",
    }
  }
>
  <div
    className="css-nen11g-MuiStack-root"
  >
    <div
      data-testid="heightBox"
      style={
        Object {
          "height": 10,
        }
      }
    />
    <div
      style={
        Object {
          "justifyContent": "center",
          "paddingLeft": 40,
        }
      }
    >
      <div
        className="css-111rfow-MuiStack-root"
      >
        <div>
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium css-zcbmsk-MuiButtonBase-root-MuiButton-root"
            data-testid="toggleBtn"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            style={
              Object {
                "textTransform": "none",
              }
            }
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="StopIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M6 6h12v12H6z"
                />
              </svg>
            </span>
            Stop Monitoring
          </button>
        </div>
        <div>
          <button
            aria-haspopup="true"
            className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
            disabled={false}
            id="basic-button"
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="SettingsIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19.14 12.94c.04-.3.06-.61.06-.94 0-.32-.02-.64-.07-.94l2.03-1.58c.18-.14.23-.41.12-.61l-1.92-3.32c-.12-.22-.37-.29-.59-.22l-2.39.96c-.5-.38-1.03-.7-1.62-.94l-.36-2.54c-.04-.24-.24-.41-.48-.41h-3.84c-.24 0-.43.17-.47.41l-.36 2.54c-.59.24-1.13.57-1.62.94l-2.39-.96c-.22-.08-.47 0-.59.22L2.74 8.87c-.12.21-.08.47.12.61l2.03 1.58c-.05.3-.09.63-.09.94s.02.64.07.94l-2.03 1.58c-.18.14-.23.41-.12.61l1.92 3.32c.12.22.37.29.59.22l2.39-.96c.5.38 1.03.7 1.62.94l.36 2.54c.05.24.24.41.48.41h3.84c.24 0 .44-.17.47-.41l.36-2.54c.59-.24 1.13-.56 1.62-.94l2.39.96c.22.08.47 0 .59-.22l1.92-3.32c.12-.22.07-.47-.12-.61l-2.01-1.58zM12 15.6c-1.98 0-3.6-1.62-3.6-3.6s1.62-3.6 3.6-3.6 3.6 1.62 3.6 3.6-1.62 3.6-3.6 3.6z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "paddingLeft": 40,
          "paddingRight": 40,
        }
      }
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-5 css-1qgewj0-MuiGrid-root"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-auto css-1wrgmsj-MuiGrid-root"
        >
          <img
            alt="CCTV video"
            height={240}
            onError={[Function]}
            src="http://localhost:5000/video_feed/cam1"
            width={320}
          />
          <div
            className="css-aj204s-MuiStack-root"
          >
            <p
              style={
                Object {
                  "color": "#6C63FF",
                  "fontSize": 20,
                  "fontWeight": 800,
                  "margin": 0,
                  "textAlign": "center",
                }
              }
            >
              CCTV - 1
            </p>
            <div
              style={
                Object {
                  "justifyContent": "center",
                  "marginLeft": "auto",
                  "marginRight": "auto",
                }
              }
            >
              <button
                className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium css-zcbmsk-MuiButtonBase-root-MuiButton-root"
                data-testid="toggleBtn"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                style={
                  Object {
                    "textTransform": "none",
                  }
                }
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="StopIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M6 6h12v12H6z"
                    />
                  </svg>
                </span>
                Stop Monitoring
              </button>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-auto css-1wrgmsj-MuiGrid-root"
        >
          <img
            alt="CCTV video"
            height={240}
            onError={[Function]}
            src="http://localhost:5000/video_feed/cam2"
            width={320}
          />
          <div
            className="css-aj204s-MuiStack-root"
          >
            <p
              style={
                Object {
                  "color": "#6C63FF",
                  "fontSize": 20,
                  "fontWeight": 800,
                  "margin": 0,
                  "textAlign": "center",
                }
              }
            >
              CCTV - 2
            </p>
            <div
              style={
                Object {
                  "justifyContent": "center",
                  "marginLeft": "auto",
                  "marginRight": "auto",
                }
              }
            >
              <button
                className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium css-zcbmsk-MuiButtonBase-root-MuiButton-root"
                data-testid="toggleBtn"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                style={
                  Object {
                    "textTransform": "none",
                  }
                }
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="StopIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M6 6h12v12H6z"
                    />
                  </svg>
                </span>
                Stop Monitoring
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      data-testid="heightBox"
      style={
        Object {
          "height": 10,
        }
      }
    />
  </div>
</div>
`;
